app/controllers/payments_controller.rb

```diff
@@ -103,7 +103,7 @@ def sent
   private
 
   def mark_payment_completed
-    Payment.transaction do
```

Comment:
No need for begin if it follows method entry. You can add rescue to any method without a begin.